docs/ci.md

```diff
@@ -54,6 +54,92 @@ Received 0 of 614674276 (0.0%), 0.0 MBs/sec
+
+#### How it works
+
+Grossly simplified, TCC works by setting up an agent which will act as Docker daemon, but proxy requests to remote VMs. It will then write a [Testcontainers configuration file](https://www.testcontainers.org/features/configuration/) that the Testcontainers library will automatically use when a test is ran.
+
+> Note that AtomicJar specifically mentions that TCC is not a Docker-as-a-Service product, and shouldn't be treated as such.
+
+Make sure to join the private Slack channel Camunda has with AtomicJar around TCC. There you will find a private link to our dashboard which contains documentation around TCC, and can provide feedback, ask qeustions, and receive support.
+
+> If you are not yet in the channel, anybody from the Zeebe team can invite you.
+
+#### Usage
+
+To use TCC, we first need to define a few environment variables to your job.
+
+```yaml
+env:
+  TC_CLOUD_LOGS_VERBOSE: true
+  TC_CLOUD_CONCURRENCY: 4
+  ZEEBE_TEST_DOCKER_IMAGE: localhost:5000/camunda/zeebe:current-test
+```
+
+Let's break this down a bit.
+
+1. Ensure the agent logs are verbose. TCC is still a beta product, so we want to make sure we can provide as much feedback as possible when troubleshooting any issues.
+1. Define how many concurrent remote VMs our tests can use to start containers on.
+1. Specify the name of the image our container based tests should use.
+
+You'll notice the image name starts with a URL, because we'll be using a local Docker registry. This is required because we want to use the image that we build as part of the pipeline, without pushing it to any public Docker registry. By using a local registry and exposing it to TCC, we can use "private" images.
+
+To do so, we'll use a [GitHub Actions service container](https://docs.github.com/en/actions/using-containerized-services/about-service-containers), and add the following to your job definition:
+
+```yaml
+services:
+  registry:
+    image: registry:2
+    ports:
+      - 5000:5000
+```
+
+With the registry set up, the next step is to push the Docker image we're building to it.
+
+```yaml
+- uses: ./.github/actions/build-docker
+  with:
+    repository: localhost:5000/camunda/zeebe
+    version: current-test
+    push: true
+    distball: ${{ steps.build-zeebe.outputs.distball }}
+```
+
+Finally, we set up the TCC agent. This can be added as a step in any workflow, provided the `TCC_CLOUD_TOKEN` secret is made available to the job.
+
+```yaml
+- name: Prepare Testcontainers Cloud agent
+  if: env.TC_CLOUD_TOKEN != ''
+  run: |
+    curl -L -o agent https://app.testcontainers.cloud/download/testcontainers-cloud-agent_linux_x86-64
```

Comment:
I couldn't find anywhere in their docs a specific version link. They do version their agent, so it's probably do-able. I think their new recommendation is to use their new GitHub action to do this, but it doesn't support using private registries so we can't use it yet.

Do you see an issue to not fix the version?

Reply:
I guess it could fail at somepoint and we first need to find out why but yeah idk

Reply:
It's kind of a double edged sword. Since it's in beta, it's good to stay up to date, but I get that, since it's in beta (:smile:), we are more likely to run into issues. :shrug: 

They also don't provide links to specific versioned binaries, though if we asked, I'm sure they would. I don't see a big issue with this now, but if you'd rather I can ask them for it.

BTW, they're working on a reusable GitHub action, so in the future I'm hoping we can just use that and version that one (and not the agent directly).